src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1843,6 +1843,7 @@ private static function processAttachmentUrls(array $urls): array
 	 * Fetches data from the object part of an activity
 	 *
 	 * @param array $object
+	 * @param bool  $c2s    "true" = The object is a "client to server" object
```

Comment:
Please link somewhere to the documentation describing these objects. If there doesn't exist a public documentation page, please be as descriptive as possible. I'm guessing you developed this feature hands-on with actual messages sent by an app, so I'd rather not lose this knowledge you acquired for the people who will have to understand this code next.

I'm still not sure what C2S is about.

Reply:
AFAIK there is no deep dive documentation. Just like AP S2S, the basic rules for AP C2S are set and the rest has to be done via the implementation.

The switch does two things:
1. C2S messages don't have an id. This will be added in the later process. So any check for the id mustn't be done here
2. The switch also defines, if we are the receiver or the sender of the message. Depending on this, we have to assign users or contacts to the actors in `to`, `cc`, `bcc` or `bto`.

Reply:
Ah wait, I think I get it. C2S is a generic ActivityPub API, overlapping in features with the Mastodon API which has become the de facto standard even though it was meant for Mastodon only.

And it uses the same endpoints as S2S? Is there a way we can do the C2S/S2S differentiation in the modules instead of the ActivityPub library?

Reply:
Yeah, it uses the same endpoints as S2S. There will also always be some overlapping in the message processing. But we should be able to move some functionality in a new C2S class. But it still would be more in the `Protocol` than the `API` namespace, I think.

Reply:
The namespace isn't important for me, I just would prefer C2S to be separate from the generic ActivityPub processing.

Reply:
Absolutely, as long as it's separate.

Reply:
Later I will move the C2S inbox stuff to there as well. But this will be done in some future PR, I think.